source/appModules/outlook.py

```diff
@@ -477,6 +479,16 @@ def _get_name(self):
 			return super(UIAGridRow, self).name
 		for index in xrange(cachedChildren.length):
 			e=cachedChildren.getElement(index)
+			UIAControlType=e.cachedControlType
+			UIAClassName=e.cachedClassName
+			# We only want to include particular children.
+			if not (
+				# Any text element should be presented.
+				UIAControlType==UIAHandler.UIA_TextControlTypeId
+				# Certain other fields that have a welldefined UIA className can also be included.
+				or UIAClassName in ("CategoryField","FlagField")
+			):
+				continue
 			name=e.cachedName

```

Comment:
In an earlier version of #7949, I used this code snippet below this line:

```
			isFlag = UIAClassName=="FlagField"
			if isFlag:
				flagIcon=0
				if selection:
					try:
						flagIcon=selection.flagIcon
					except COMError:
						pass
				if not flagIcon:
					# This message has no flag.
					# Filter the flag field, as it provides redundant information.
					# Ignore flags if the object model couldn't be accessed.
					continue
```

Reply:
THen, below the `if text:` line:

```
				if isFlag:
					# At least Outlook 2016 seems to have a bug where it is not possible to reorder the flag field in a view.
					# Since flags are considered important, announce them as early as possible.
					textList.insert(0, text)
				else:
					text+=u","
					textList.append(text)
```

Feel free to discard from this code whatever you do not like. And at least do not forget to remove the flag fetching that happens earlier :)